presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/TableScanMatcher.java

```diff
@@ -64,35 +63,8 @@ public MatchResult detailMatches(PlanNode node, PlanNodeCost cost, Session sessi
         String actualTableName = tableMetadata.getTable().getTableName();
         return new MatchResult(
                 expectedTableName.equalsIgnoreCase(actualTableName) &&
-                        domainMatches(tableScanNode, session, metadata));
-    }
-
-    private boolean domainMatches(TableScanNode tableScanNode, Session session, Metadata metadata)
-    {
-        if (!expectedConstraint.isPresent()) {
-            return true;
-        }
-
-        TupleDomain<ColumnHandle> actualConstraint = tableScanNode.getCurrentConstraint();
-        if (expectedConstraint.isPresent() && !actualConstraint.getDomains().isPresent()) {
-            return false;
-        }
-
-        Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableScanNode.getTable());
-        for (Map.Entry<String, Domain> expectedColumnConstraint : expectedConstraint.get().entrySet()) {
-            if (!columnHandles.containsKey(expectedColumnConstraint.getKey())) {
-                return false;
-            }
-            ColumnHandle columnHandle = columnHandles.get(expectedColumnConstraint.getKey());
-            if (!actualConstraint.getDomains().get().containsKey(columnHandle)) {
-                return false;
-            }
-            if (!expectedColumnConstraint.getValue().contains(actualConstraint.getDomains().get().get(columnHandle))) {
-                return false;
-            }
-        }
-
-        return true;
+                        ((!expectedConstraint.isPresent()) ||
```

Comment:
If it was in a separate commit, the factored-out function would have only one caller.